src/synth/fluid_synth.c

```diff
@@ -595,8 +593,10 @@ new_fluid_synth(fluid_settings_t *settings)
   
   synth->settings = settings;
 
-  fluid_settings_getint(settings, "synth.reverb.active", &synth->with_reverb);
-  fluid_settings_getint(settings, "synth.chorus.active", &synth->with_chorus);
+  fluid_settings_getint(settings, "synth.reverb.active", &with_reverb);
+  fluid_atomic_int_set(&synth->with_reverb, with_reverb);  
+  fluid_settings_getint(settings, "synth.chorus.active", &with_chorus);
+  fluid_atomic_int_set(&synth->with_chorus, with_chorus);  
```

Comment:
Again: Constructor: dont see necessity.

Reply:
atomic type need to be opaque

Reply:
I actually quite like this change and also the fluid_atomic_* typedefs. They make it immediately obvious that these values are meant to be accessed atomically throughout the codebase.

Reply:
They might even help us to get rid of these annoying gcc warnings:
```
/usr/include/glib-2.0/glib/gatomic.h:117:5: warning: cast from function call of type ‘_Bool’ to non-matching type ‘int’ [-Wbad-function-cast]
```

Reply:
Maybe. However it should be `typedef fluid_atomic_int_t`. And it should also cleanly be done in one commit only addressing this issue and not something else.

Reply:
so you want me to remake splited pull requests ?